test/CMakeLists.txt

```diff
@@ -28,111 +22,36 @@ find_package(
-
-# options common to both fftw/mkl
-set(LINK_LIBRARIES_COMMON
-    ${NEKTAR++_LIBRARIES} ${NEKTAR++_TP_LIBRARIES} ${Boost_LIBRARIES}
-    ${MPI_CXX_LINK_FLAGS} ${MPI_CXX_LIBRARIES} GTest::gtest)
-
-set(INCLUDE_DIRECTORIES_COMMON ${CMAKE_SOURCE_DIR}/include
-                               ${Boost_INCLUDE_DIRS} ${MPI_CXX_INCLUDE_PATH})
-
-# Build the tests individually
-include(GoogleTest)
-foreach(TEST ${TEST_SRCS})
-  get_filename_component(TEST_NAME ${TEST} NAME_WLE)
-  message(STATUS "Found test - ${TEST_NAME}")
-  set(TEST_LIST ${TEST_LIST} ${TEST_NAME})
-
-  set(TEST_SOURCES ${TEST_MAIN} ${TEST})
-  add_executable(${TEST_NAME} ${TEST_SOURCES})
-  include_directories(${Boost_INCLUDE_DIRS})
-
-  if(MKL_FOUND)
-    target_link_libraries(
-      ${TEST_NAME} PRIVATE ${LIBRARY_NAME} ${CUSTOM_MKL_LINK}
-                           ${LINK_LIBRARIES_COMMON})
-    target_include_directories(
-      ${TEST_NAME}
-      PRIVATE $<TARGET_PROPERTY:MKL::MKL,INTERFACE_INCLUDE_DIRECTORIES>
-              ${INCLUDE_DIRECTORIES_COMMON})
-
-    target_compile_options(
-      ${TEST_NAME} PUBLIC -fsycl
-                          $<TARGET_PROPERTY:MKL::MKL,INTERFACE_COMPILE_OPTIONS>)
-  endif()
-
-  if(FFTW_FOUND)
-    target_link_libraries(${TEST_NAME} PRIVATE ${LIBRARY_NAME} PkgConfig::FFTW
-                                               ${LINK_LIBRARIES_COMMON})
-    target_include_directories(
-      ${TEST_NAME} PRIVATE PkgConfig::FFTW ${INCLUDE_DIRECTORIES_COMMON})
-  endif()
-
-  add_sycl_to_target(TARGET ${TEST_NAME} SOURCES ${TEST_SOURCES})
-  gtest_add_tests(TARGET ${TEST_NAME} SOURCES ${TEST_SOURCES})
-endforeach()
-
-# Build a global test suite
-add_executable(${EXECUTABLE} ${TEST_MAIN} ${TEST_LIST})
-
-if(MKL_FOUND)
-  target_compile_options(
-    ${EXECUTABLE} PUBLIC $<TARGET_PROPERTY:MKL::MKL,INTERFACE_COMPILE_OPTIONS>)
-  target_include_directories(
-    ${EXECUTABLE}
-    PRIVATE $<TARGET_PROPERTY:MKL::MKL,INTERFACE_INCLUDE_DIRECTORIES>
-            ${INCLUDE_DIRECTORIES_COMMON})
-  target_link_libraries(
-    ${EXECUTABLE} PRIVATE ${LIBRARY_NAME} ${CUSTOM_MKL_LINK}
-                          ${LINK_LIBRARIES_COMMON})
-
-  target_compile_options(
-    ${EXECUTABLE} PUBLIC -fsycl
-                         $<TARGET_PROPERTY:MKL::MKL,INTERFACE_COMPILE_OPTIONS>)
-endif()
-
-if(FFTW_FOUND)
-  target_include_directories(
-    ${EXECUTABLE} PRIVATE PkgConfig::FFTW ${INCLUDE_DIRECTORIES_COMMON})
-  target_link_libraries(${EXECUTABLE} PRIVATE ${LIBRARY_NAME} PkgConfig::FFTW
-                                              ${LINK_LIBRARIES_COMMON})
-endif()
-
-# define the test executable as a sycl target
-add_sycl_to_target(TARGET ${EXECUTABLE} SOURCES ${TEST_MAIN} ${TEST_LIST}
-                   ${SRC_LIST})
-gtest_add_tests(TARGET ${EXECUTABLE} SOURCES ${TEST_MAIN})
+set(UNIT_EXE unitTests)
+set(UNIT_SRC ${CMAKE_CURRENT_SOURCE_DIR}/unit)
+set(UNIT_SRC_FILES
+    ${UNIT_SRC}/test_diagnostics.cpp
```

Comment:
According to the [CMake documentation](https://cmake.org/cmake/help/latest/command/file.html#glob):

>Note: We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Reply:
Just to be clear, the CMakeLists.txt will need to be updated to add the names of specific tests.

Reply:
Yes. Up to you whether you think it's best to stick with the name globbing anyway to avoid that.